### ISSUE TITLE
Not using cross entropy style loss when training with Mix(Point(),Box())

### ISSUE THREAD

**Issue by ForeverZyh** [eth-sri/diffai#7]
I trained with the following command:
``python . -d "Mix(a=Point(),b=Box(),aw=1,bw=0.2)" -t "Point()" -n ffnn``
Ideally, it will using the cross entropy style loss for the Box domain as suggested by Gowal et al. 2019, but it turns out that it still used the original loss in DiffAI paper.

I looked through the implementation of ``loss`` in ai.py and goals.py. It seems the cross entropy style loss only works in training with `-d "Box()"` but does not work with Mix Domain. I don't know if this behavior is desired.

**Comment by mmirman** [comment]
This is very strange, it should definitely be using the crossentropy style loss.  If it actually is using the original loss, this would be very interesting as the results show significant improvement over those from the original paper.  I will look into this.

**Comment by mmirman** [comment]
After print statement instrumentation, I've found no evidence of this occurring with the training domain you mentioned.    Its worth noting that the command you provided's neural network isn't the current name, so perhaps you are still on an old commit?

**Comment by ForeverZyh** [comment]
I think I made a stupid mistake: after modifying your code, I lost the `tag` in `TaggedDomain` somewhere so that it did not train with cross entropy style loss.
BTW, the command is modified from the README, I also noticed that I should change the `ffnn` to `FFNN`.
Close the issue.
